src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs

```diff
@@ -24,20 +24,31 @@ public static class TreeExtensions
         /// <param name="numLeaves">The maximum number of leaves per decision tree.</param>
         /// <param name="minDatapointsInLeaves">The minimal number of datapoints allowed in a leaf of a regression tree, out of the subsampled data.</param>
         /// <param name="learningRate">The learning rate.</param>
-        /// <param name="advancedSettings">Algorithm advanced settings.</param>
         public static FastTreeRegressionTrainer FastTree(this RegressionContext.RegressionTrainers ctx,
             string labelColumn = DefaultColumnNames.Label,
             string featureColumn = DefaultColumnNames.Features,
             string weights = null,
             int numLeaves = Defaults.NumLeaves,
             int numTrees = Defaults.NumTrees,
             int minDatapointsInLeaves = Defaults.MinDocumentsInLeaves,
-            double learningRate = Defaults.LearningRates,
-            Action<FastTreeRegressionTrainer.Arguments> advancedSettings = null)
+            double learningRate = Defaults.LearningRates)
+        {
+            Contracts.CheckValue(ctx, nameof(ctx));
+            var env = CatalogUtils.GetEnvironment(ctx);
+            return new FastTreeRegressionTrainer(env, labelColumn, featureColumn, weights, numLeaves, numTrees, minDatapointsInLeaves, learningRate);
+        }
+
+        /// <summary>
+        /// Predict a target using a decision tree regression model trained with the <see cref="FastTreeRegressionTrainer"/>.
+        /// </summary>
+        /// <param name="ctx">The <see cref="RegressionContext"/>.</param>
+        /// <param name="advancedSettings">Algorithm advanced settings.</param>
+        public static FastTreeRegressionTrainer FastTree(this RegressionContext.RegressionTrainers ctx,
+            FastTreeRegressionTrainer.Options advancedSettings)
```

Comment:
>advancedSettings [](start = 46, length = 16)

I agree that we should name this something different from arguments. However, now it does not only contain advanced settings, it also contains the names of the input columns, and some basic settings. Would be great if this name matched the name we give to the Options parameter in the public constructor of the trainers.